week10/pages/Register.jsx

```diff
@@ -0,0 +1,89 @@
+import { useState } from "react";
+import { postRegister } from "../api/user";
+import { postLogin } from "../api/user";
+import { useNavigate } from "react-router-dom";
+const Register = () => {
+  const [signup, setSignup] = useState({ id: "", pw: "", nickname: "" });
+  const [login, setLogin] = useState({ id: "", pw: "" });
+  const navigate = useNavigate();
+  const onChangeSignup = (e) => {
+    const { name, value } = e.target;
+    setSignup({ ...signup, [name]: value });
+  };
+  const onChangeLogin = (e) => {
+    const { name, value } = e.target;
+    setLogin({ ...login, [name]: value });
+  };
+  const Signup = async (e) => {
+    e.preventDefault();
+    const { id, pw, nickname } = signup;
+    const res = await postRegister(id, pw, nickname);
+    if (res.status === 201) {
+      alert("회원가입 성공! 로그인을 진행해주세요.");
+      setSignup({ id: "", pw: "", nickname: "" });
+    } else if (res.response.status === 400) {
+      alert(res.response.data.message);
+    }
+  };

```

Comment:
P5: 에러처리를 할 때는 try-catch 블록을 활용해서 예상치 못한 상황을 미리 방지하면 좋습니다. 예시 코드를 참고하시면 Signup 함수와 Login 함수에 모두 적용할 수 있을 것 같아요!
```suggestion
const Signup = async (e) => {
  e.preventDefault();
  const { id, pw, nickname } = signup;
  try {
    const res = await postRegister(id, pw, nickname);
    if (res.status === 201) {
      alert("회원가입 성공! 로그인을 진행해주세요.");
      setSignup({ id: "", pw: "", nickname: "" });
    } else {
      throw new Error(`Unexpected response status: ${res.status}`);
    }
  } catch (err) {
    if (err.response && err.response.status === 400) {
      alert(err.response.data.message);
    } else {
      console.error(err);
      alert("회원가입 오류 메세지");
    }
  }
};

```